src/sentry/eventstore/models.py

```diff
@@ -362,33 +362,52 @@ def get_hashes(self, force_config=None):
             if hashes is not None:
                 return hashes, hierarchical_hashes
 
+        # Create fresh hashes
+        flat_variants, hierarchical_variants = self.get_sorted_grouping_variants(force_config)
+        flat_hashes = self._hashes_from_sorted_grouping_variants(flat_variants)
+        hierarchical_hashes = self._hashes_from_sorted_grouping_variants(hierarchical_variants)
+
+        return flat_hashes, hierarchical_hashes
+
+    def get_sorted_grouping_variants(self, force_config=None):
+        """ Get grouping variants sorted into flat and hierarchical variants """
+        variants = self.get_grouping_variants(force_config)
+        return self._sort_grouping_variants(variants)
+
+    @staticmethod
+    def _hashes_from_sorted_grouping_variants(variants):
+        """ Create hashes from variants and filter out None values """
+        hashes = (variant.get_hash() for variant in variants)
+        return [hash_ for hash_ in hashes if hash_ is not None]
+
+    @staticmethod
+    def _sort_grouping_variants(variants):
```

Comment:
@untitaker should I move this function into a different module, as it does not depend on the Event instance?

Reply:
I would put the sorting fn into `sentry.grouping` but not the rest

Ideally we would actually find an API that does not require eventstore.models to call more than one function from sentry.grouping.